### ISSUE TITLE
Special characters in image file names

### ISSUE THREAD

**Issue by GitHub-Mike** [janreges/siteone-crawler#33]
Some images were not saved because the file name contains special characters. These are spaces and German umlauts, but other special characters will certainly also cause problems.

I would suggest using URL Encoding (Percent-Encoding) according to [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986) to store file names. This should cause the fewest problems with the most common operating systems and file systems.

What do you think?

**Comment by GitHub-Mike** [comment]
> please provide sample URLs directly from the HTML code of the source website. I need to know if this is related to query strings and the recently added option not to hash query strings.

No, the problem has existed since my first crawl on 06.12.2024 and has nothing to do with Issue #30.

Here are 2 examples:

```
- "/images/2021/04/verkauf-april-2021 verschoben.jpg" --> /images/2021/04/verkauf-april-2021%20verschoben.jpg
- "/images/2021/03/verkauf-märz-2021.jpg" --> /images/2021/03/verkauf-m%C3%A4rz-2021.jpg
```

> Alternatively, if you can, please provide the URL to the tested website. This would be very helpful for me to be able to debug this on a specific site. Thanks.

I don't want to publish the URL of the website here, but I can send it to you by e-mail.

**Comment by nilocky** [comment]
I got similar issue and the crawler treated the links with space in `<a>` tag as 404 URLs.

Example below:

**URL parsed in the 404 URLs report**
`/download/announcement/22-3-2023`

**Actual link in the page under `<a>` tag**
`/download/announcement/22-3-2023 file has space in filename.pdf`
